rllib/algorithms/impala/impala.py

```diff
@@ -645,7 +715,34 @@ def default_resource_request(
                 ]
                 if cf.evaluation_interval
                 else []
-            ),
+            )
+        )
+        if cf._enable_learner_api:
+            # resources for the trainer
+            if cf.num_learner_workers == 0:
+                # if num_learner_workers is 0, then we need to allocate one gpu if
+                # num_gpus_per_learner_worker is greater than 0.
+                trainer_bundle = [
+                    {
+                        "CPU": cf.num_cpus_per_learner_worker,
+                        "GPU": cf.num_gpus_per_learner_worker,
+                    }
+                ]
+            else:
+                trainer_bundle = [
```

Comment:
not inheriting from base class's function, so we need to add a pg bundle for the learner group